receiver/oracledbreceiver/scraper.go

```diff
@@ -430,6 +497,248 @@ func (s *oracleScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	return out, nil
 }
 
+type queryMetricCacheHit struct {
+	sqlID        string
+	childNumber  string
+	childAddress string
+	queryText    string
+	metrics      map[string]int64
+}
+
+func (s *oracleScraper) scrapeLogs(ctx context.Context) (plog.Logs, error) {
+	var scrapeErrors []error
+
+	if s.logsBuilderConfig.Events.DbServerTopQuery.Enabled {
+		topNLogs, topNCollectionErrors := s.collectTopNMetricData(ctx)
+		if topNCollectionErrors != nil {
+			scrapeErrors = append(scrapeErrors, topNCollectionErrors)
+		} else {
+			return topNLogs, errors.Join(scrapeErrors...)
+		}
+	}
+
+	return plog.NewLogs(), errors.Join(scrapeErrors...)
+}
+
+func (s *oracleScraper) collectTopNMetricData(ctx context.Context) (plog.Logs, error) {
+	var errs []error
+	// get metrics and query texts from DB
+	timestamp := pcommon.NewTimestampFromTime(time.Now())
+	intervalSeconds := int(s.scrapeCfg.CollectionInterval.Seconds())
+	s.oracleQueryMetricsClient = s.clientProviderFunc(s.db, oracleQueryMetricsSQL, s.logger)
+	now := timestamp.AsTime().Format(dbTimeReferenceFormat)
+	metricRows, metricError := s.oracleQueryMetricsClient.metricRows(ctx, now, intervalSeconds, s.topQueryCollectCfg.MaxQuerySampleCount)
+
+	if metricError != nil {
+		errs = append(errs, fmt.Errorf("error executing %s: %w", oracleQueryMetricsSQL, metricError))
+		return plog.NewLogs(), errors.Join(errs...)
+	}
+
+	if len(metricRows) == 0 {
+		errs = append(errs, errors.New("no data returned from oracleQueryMetricsClient"))
+		return plog.NewLogs(), errors.Join(errs...)
```

Comment:
oracleQueryMetricsSQL is probably too big and not very useful to put in the stdout
```suggestion
		return plog.NewLogs(), fmt.Errorf("error executing oracleQueryMetricsSQL: %w", metricError)
	}

	if len(metricRows) == 0 {
		return plog.NewLogs(), errors.New("no data returned from oracleQueryMetricsClient")
```